### PR TITLE
feat: add helm repo install check

### LINKED ISSUE #750: add test for helm repo install
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

related to https://github.com/kubernetes-sigs/blob-csi-driver/issues/746, we need to write a test for helm repo install issue.

or provide a test to check all links inside https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/charts/index.yaml directly.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/feature

**What this PR does / why we need it**:

add test for helm repo install

**Which issue(s) this PR fixes**:

Fixes #750

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
use `sh ./test/utils/check_helm_repo.sh` in dir to check url in index.yaml

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
pls add this file to ./hack/ folder, and then add this script in verify-all.sh file

File: hack/verify-helm-chart-index.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    else
+        echo "$1 is fail."
+        echo "rm "${PKG_ROOT}${url#*master}
+        rm -rf ${PKG_ROOT}${url#*master}
+        echo "helm repo index"
+        helm repo index ${PKG_ROOT}/charts/ --url https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/
+    fi
+}
+
+check_yaml() {
```
Comment (andyzhangx) [inline hack/verify-helm-chart-index.sh:38]: that makes logic too complex, what about this way:
```
# grep http index.yaml
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.16.0/blob-csi-driver-v1.16.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.15.0/blob-csi-driver-v1.15.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.14.0/blob-csi-driver-v1.14.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.13.0/blob-csi-driver-v1.13.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.12.0/blob-csi-driver-v1.12.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.11.0/blob-csi-driver-v1.11.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.10.0/blob-csi-driver-v1.10.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.9.0/blob-csi-driver-v1.9.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.8.0/blob-csi-driver-v1.8.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.7.0/blob-csi-driver-v1.7.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.6.0/blob-csi-driver-v1.6.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.5.0/blob-csi-driver-v1.5.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.4.1/blob-csi-driver-v1.4.1.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.4.0/blob-csi-driver-v1.4.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.3.0/blob-csi-driver-v1.3.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.2.0/blob-csi-driver-v1.2.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.1.0/blob-csi-driver-v1.1.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/v1.0.0/blob-csi-driver-v1.0.0.tgz
    - https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts/latest/blob-csi-driver-v0.0.0.tgz
```

File: hack/verify-helm-chart-index.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+INDEX=${PKG_ROOT}/charts/index.yaml
+
+function check_url() {  
+    url=$1
+    result=$(curl -I -m 5 -s -w "%{http_code}\n" -o /dev/null $1)
+    if [ $result -ne 200 ]
+    then
+        echo "Error: $1 is fail."
+        echo "Error: wrong pkg in "${PKG_ROOT}${url#*master}
```
Comment (andyzhangx) [inline hack/verify-helm-chart-index.sh:29]: add `exit 1` to make sure if there is invalid link, this script should fail

Comment (umagnus) [reply hack/verify-helm-chart-index.sh:29]: ok, it is added.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**